### ISSUE TITLE
False positive on code involving typevar with default

### ISSUE THREAD

**Issue by luwqz1** [microsoft/pyright#9246]
```python
import typing

T = typing.TypeVar("T", bound="Base", default="Foo")


class Base: ...


class Foo(Base): ...


class Bar(typing.Generic[T]):
    def __init__(self, foo: T) -> None:
        self.foo = foo

    @property
    def foo_(self) -> T:
        return self.foo  # error
```

But mypy works correctly [mypy playground](https://mypy-play.net/?mypy=latest&python=3.13&gist=2667524bdd0cc7cb1eea42eed81a537f)

This is probably a regression because there was no error on version 1.1.383

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
